### DISCUSSION TITLE
[Bug report] Wrong domain shown at the top of the page

### DISCUSSION THREAD

**Discussion by Ark-kun** [community/community#23557] [Other Feature Feedback, Questions, & Ideas]
<p>I use GitHub Pages with a custom domain (<a href="https://cloud-pipelines.net/" rel="noopener nofollow ugc">https://cloud-pipelines.net/</a>). I’ve added a Markdown file to the repository. When I visit the page corresponding to that file, I see the old GitHub Pages domain (<a href="http://cloud-pipelines.github.io" rel="noopener nofollow ugc">cloud-pipelines.github.io</a>) shown at the top of the page instead of the custom domain. The same domain is used for the the page title (tab title). See <a href="https://cloud-pipelines.net/privacy_policy" class="inline-onebox" rel="noopener nofollow ugc">Privacy policy | cloud-pipelines.github.io</a></p>
<p>Here is the list of the places where the wrong domain is used:</p>
<ul>
<li><code>&lt;title&gt;Privacy policy | cloud-pipelines.github.io&lt;/title&gt;</code></li>
<li><code>&lt;meta property="og:site_name" content="cloud-pipelines.github.io" /&gt;</code></li>
<li><code>&lt;h1&gt;&lt;a href="https://cloud-pipelines.net/"&gt;cloud-pipelines.github.io&lt;/a&gt;&lt;/h1&gt;</code></li>
</ul>
<p>Could you please fix this issue, so that the custom domain is used everywhere consistently?</p>
<p>HTML snippet:</p>
<pre><code class="lang-auto">&lt;!-- Begin Jekyll SEO tag v2.7.1 --&gt;
&lt;title&gt;Privacy policy | cloud-pipelines.github.io&lt;/title&gt;
&lt;meta name="generator" content="Jekyll v3.9.0" /&gt;

...

&lt;link rel="canonical" href="https://cloud-pipelines.net/privacy_policy.html" /&gt;
&lt;meta property="og:url" content="https://cloud-pipelines.net/privacy_policy.html" /&gt;
&lt;meta property="og:site_name" content="cloud-pipelines.github.io" /&gt;

...
  &lt;/head&gt;
  &lt;body&gt;
    &lt;div class="container-lg px-3 my-5 markdown-body"&gt;
      
      &lt;h1&gt;&lt;a href="https://cloud-pipelines.net/"&gt;cloud-pipelines.github.io&lt;/a&gt;&lt;/h1&gt;
</code></pre>

**Comment by yoannchaudet** [comment]
<p>Hello!</p>
<p>I don’t see any references of <a href="http://github.io">github.io</a> on your site now. Is it resolved?</p>
<p>It could have been caching.</p>

**Comment by Ark-kun** [comment]
<p>Thank you for looking into this.</p>
<blockquote>
<p>don’t see any references of <a href="http://github.io/" rel="noopener nofollow ugc">github.io</a> on your site now. Is it resolved?</p>
</blockquote>
<p>I still see <a href="http://github.io" rel="noopener nofollow ugc">github.io</a> in all 3 places on the Privacy page. See screenshot. <a href="https://cloud-pipelines.net/privacy_policy" class="inline-onebox" rel="noopener nofollow ugc">Privacy policy | cloud-pipelines.github.io</a></p>
<p>It’s even visible in my original post - GitHub community link title generator fetched the page and the title had “<a href="http://github.io" rel="noopener nofollow ugc">github.io</a>”. See the 2nd screenshot.</p>
<blockquote>
<p>It could have been caching.</p>
</blockquote>
<p>I’ve switched to custom domain several months ago and the privacy policy doc was created long after the switch. So it cannot be cache…</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/d/f/dfd83201276786ade0b19bcd50b7662b549dd604.png" data-download-href="/uploads/short-url/vWdFGcrx6QenNGxh0SaGvwcW2q0.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1829149/181077885-be3983bd-6e17-4804-925d-ef69e0371057.png" alt="image" data-base62-sha1="vWdFGcrx6QenNGxh0SaGvwcW2q0" width="534" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/d/f/dfd83201276786ade0b19bcd50b7662b549dd604_2_534x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/d/f/dfd83201276786ade0b19bcd50b7662b549dd604_2_801x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/d/f/dfd83201276786ade0b19bcd50b7662b549dd604_2_1068x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/f/dfd83201276786ade0b19bcd50b7662b549dd604_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1808×1691 268 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/f/3/f3b33980172a6a44bd52431caca9e57a4c0d4c38.png" data-download-href="/uploads/short-url/yLS0LbTafhcL26lodNIHMwrqd6M.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1829149/181077886-6e291b90-2422-430e-a013-a25ed493711c.png" alt="image" data-base62-sha1="yLS0LbTafhcL26lodNIHMwrqd6M" width="690" height="185" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/f/3/f3b33980172a6a44bd52431caca9e57a4c0d4c38_2_690x185.png, https://aws1.discourse-cdn.com/github/optimized/3X/f/3/f3b33980172a6a44bd52431caca9e57a4c0d4c38_2_1035x277.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/f/3/f3b33980172a6a44bd52431caca9e57a4c0d4c38.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/f/3/f3b33980172a6a44bd52431caca9e57a4c0d4c38_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1158×312 55.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by yoannchaudet** [comment]
<p>Oh my mistake I get what you mean now.</p>
<p>First you have two privacy pages in your repo:</p>
<ol>
<li>
<a href="https://cloud-pipelines.net/privacy" class="inline-onebox">Privacy policy | cloud-pipelines.net</a> (which is an actual <a href="https://github.com/Cloud-Pipelines/cloud-pipelines.github.io/blob/f45687e1d519364787920ed4a4b38f26a3ae1623/privacy.html">html file</a> in your repo hence not rendered by Jekyll)</li>
<li>
<a href="https://cloud-pipelines.net/privacy_policy" class="inline-onebox">Privacy policy | cloud-pipelines.github.io</a> which is a <a href="https://github.com/Cloud-Pipelines/cloud-pipelines.github.io/blob/f45687e1d519364787920ed4a4b38f26a3ae1623/privacy_policy.md">markdown file</a> in your repo hence rendered by Jekyll</li>
</ol>
<p>Since you are not specifying any Jekyll configuration, your markdown files (2) are rendered with the default theme that is <a href="https://github.com/pages-themes/primer"><code>primer</code></a>. There are a few things that come with the theme including including your site’s title in the three location you have noticed (title, meta and h1). Your site’s title is your repository name by default, <code>cloud-pipelines.github.io</code> here.</p>
<p>While you could rename your repository, a better approach here would be to simply specify the title of your site in a Jekyll configuration file. You can do that by creating a <code>_config.yml</code> at the root of your repository with the following content:</p>
<pre><code class="lang-auto">title: cloud-pipelines.net
</code></pre>

**Comment by Ark-kun** [comment]
<p>Thank you for helping!<br>
This did not occur to me for some reason. I was a bit confused about how Markdown to HTML generation works when there is no configuration. For some reason I did not realize it was the same Jekyll (yeah, I saw the comment in HTML, but…).</p>
<p>The situation looked weird, because the page generator <em>knew</em> the correct domain (the title link links to it). So, <code>&lt;a href="https://custom_domain"&gt;github_domain&lt;/a&gt;</code> looked like a bug.</p>
<p>I’ve followed your advice, specified title in <code>_config.yml</code> and everything works great.</p>
<p>Thank you again</p>
<blockquote>
<ol>
<li>
<a href="https://cloud-pipelines.net/privacy" rel="noopener nofollow ugc">Privacy policy | cloud-pipelines.net</a> (which is an actual <a href="https://github.com/Cloud-Pipelines/cloud-pipelines.github.io/blob/f45687e1d519364787920ed4a4b38f26a3ae1623/privacy.html" rel="noopener nofollow ugc">html file</a> in your repo hence not rendered by Jekyll)</li>
</ol>
</blockquote>
<p>Yes. I needed to fix the issue somehow, so I copied the generated HTML and edited it manually. I committed it as another file with different name (<code>privacy.html</code>) so that it does not interfere with the repro case in this thread.</p>
